src/sql/connection.py

```diff
@@ -369,20 +368,27 @@ def set(cls, descriptor, displaycon, connect_args=None, creator=None, alias=None
                 cls.current = CustomConnection(descriptor, alias=alias)
             else:
                 existing = rough_dict_get(cls.connections, descriptor)
-
                 # NOTE: I added one indentation level, otherwise
                 # the "existing" variable would not exist if
                 # passing an engine object as descriptor.
                 # Since I never saw this breaking, my guess
                 # is that we're missing some unit tests
                 # when descriptor is a connection object
                 # http://docs.sqlalchemy.org/en/rel_0_9/core/engines.html#custom-dbapi-connect-arguments # noqa
-                cls.current = existing or Connection.from_connect_str(
-                    connect_str=descriptor,
-                    connect_args=connect_args,
-                    creator=creator,
-                    alias=alias,
-                )
+                # if same alias found
```

Comment:
@edublancas 
Please check whether this is a desired behavior. If yes, the logic looks good, and it is working on binder